From: Hannes Reinecke <hare@suse.de>
Date: Thu, 10 Nov 2011 16:32:28 +0100
Subject: SCSI: set queuedata to NULL before calling scsi_free_queue()
Patch-Mainline: not yet
References: bnc#729548

We need to set queuedata to NULL to avoid WARN_ON() during
scanning.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/scsi/scsi_scan.c b/drivers/scsi/scsi_scan.c
index 0842889..1f395c5 100644
--- a/drivers/scsi/scsi_scan.c
+++ b/drivers/scsi/scsi_scan.c
@@ -322,6 +322,7 @@ out_device_destroy:
 	scsi_device_set_state(sdev, SDEV_DEL);
 	transport_destroy_device(&sdev->sdev_gendev);
 	put_device(&sdev->sdev_dev);
+	sdev->request_queue->queuedata = NULL;
 	scsi_free_queue(sdev->request_queue);
 	put_device(&sdev->sdev_gendev);
 out:
